homeassistant/components/blink/services.py

```diff
@@ -3,171 +3,76 @@
-    CONF_NAME,
-    CONF_PIN,
-)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import CONF_PIN
 from homeassistant.core import HomeAssistant, ServiceCall
 from homeassistant.exceptions import HomeAssistantError, ServiceValidationError
 import homeassistant.helpers.config_validation as cv
-import homeassistant.helpers.device_registry as dr
 
-from .const import (
-    DOMAIN,
-    SERVICE_REFRESH,
-    SERVICE_SAVE_RECENT_CLIPS,
-    SERVICE_SAVE_VIDEO,
-    SERVICE_SEND_PIN,
-)
-from .coordinator import BlinkUpdateCoordinator
+from .const import ATTR_CONFIG_ENTRY_ID, DOMAIN, SERVICE_REFRESH, SERVICE_SEND_PIN
 
 SERVICE_UPDATE_SCHEMA = vol.Schema(
     {
-        vol.Required(ATTR_DEVICE_ID): vol.All(cv.ensure_list, [cv.string]),
-    }
-)
-SERVICE_SAVE_VIDEO_SCHEMA = vol.Schema(
-    {
-        vol.Required(ATTR_DEVICE_ID): vol.All(cv.ensure_list, [cv.string]),
-        vol.Required(CONF_NAME): cv.string,
-        vol.Required(CONF_FILENAME): cv.string,
+        vol.Required(ATTR_CONFIG_ENTRY_ID): vol.All(cv.ensure_list, [cv.string]),
     }
 )
 SERVICE_SEND_PIN_SCHEMA = vol.Schema(
     {
-        vol.Required(ATTR_DEVICE_ID): vol.All(cv.ensure_list, [cv.string]),
+        vol.Required(ATTR_CONFIG_ENTRY_ID): vol.All(cv.ensure_list, [cv.string]),
         vol.Optional(CONF_PIN): cv.string,
     }
 )
-SERVICE_SAVE_RECENT_CLIPS_SCHEMA = vol.Schema(
-    {
-        vol.Required(ATTR_DEVICE_ID): vol.All(cv.ensure_list, [cv.string]),
-        vol.Required(CONF_NAME): cv.string,
-        vol.Required(CONF_FILE_PATH): cv.string,
-    }
-)
 
 
 def setup_services(hass: HomeAssistant) -> None:
     """Set up the services for the Blink integration."""
 
-    def collect_coordinators(
-        device_ids: list[str],
-    ) -> list[BlinkUpdateCoordinator]:
-        config_entries: list[ConfigEntry] = []
-        registry = dr.async_get(hass)
-        for target in device_ids:
-            device = registry.async_get(target)
-            if device:
-                device_entries: list[ConfigEntry] = []
-                for entry_id in device.config_entries:
-                    entry = hass.config_entries.async_get_entry(entry_id)
-                    if entry and entry.domain == DOMAIN:
-                        device_entries.append(entry)
-                if not device_entries:
-                    raise ServiceValidationError(
-                        translation_domain=DOMAIN,
-                        translation_key="invalid_device",
-                        translation_placeholders={"target": target, "domain": DOMAIN},
-                    )
-                config_entries.extend(device_entries)
-            else:
-                raise HomeAssistantError(
+    async def send_pin(call: ServiceCall):
+        """Call blink to send new pin."""
+        for entry_id in call.data[ATTR_CONFIG_ENTRY_ID]:
+            try:
+                coordinator = hass.data[DOMAIN][entry_id]
+            except KeyError as ex:
```

Comment:
Check the config entry existance and state first instead. The KeyError here can have different causes.

Use `hass.config_entries.async_get_entry`.